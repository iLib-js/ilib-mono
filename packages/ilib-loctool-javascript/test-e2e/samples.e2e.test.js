/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-javascript samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require("path");
const { expectFileToMatchSnapshot, LoctoolRunner, FSSnapshot } = require("ilib-internal");

describe("samples", () => {
    describe("js", () => {
        /** @type {FSSnapshot} */
        let fsSnapshot;
        const projectPath = path.resolve(__dirname, "..", "samples", "js");
        const xliffPath = path.resolve(projectPath, "sample-js-extracted.xliff");

        beforeAll(async () => {
            fsSnapshot = FSSnapshot.create(
                [
                    "sample-js-extracted.xliff",
                    "sample-js-new-de-DE.xliff",
                    "sample-js-new-ko-KR.xliff",
                    "resources",
                ].map((p) => path.resolve(projectPath, p))
            );
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            fsSnapshot.restore();
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });

    describe("js-json", () => {
        /** @type {FSSnapshot} */
        let fsSnapshot;
        const projectPath = path.resolve(__dirname, "..", "samples", "js-json");
        const xliffPath = path.resolve(projectPath, "sample-js-json-extracted.xliff");

        beforeAll(async () => {
            fsSnapshot = FSSnapshot.create(
                [
                    "sample-js-json-extracted.xliff",
                    "sample-js-json-new-de-DE.xliff",
                    "sample-js-json-new-ko-KR.xliff",
                    "resources",
                ].map((p) => path.resolve(projectPath, p))
            );
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            fsSnapshot.restore();
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });

    describe("php-resource", () => {
        /** @type {FSSnapshot} */
        let fsSnapshot;
        const projectPath = path.resolve(__dirname, "..", "samples", "php-resource");
        const xliffPath = path.resolve(projectPath, "sample-php-resource-extracted.xliff");

        beforeAll(async () => {
            fsSnapshot = FSSnapshot.create(
                [
                    "sample-php-resource-extracted.xliff",
                    "sample-php-resource-new-de-DE.xliff",
                    "sample-php-resource-new-ko-KR.xliff",
                    "resources",
                ].map((p) => path.resolve(projectPath, p))
            );
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            fsSnapshot.restore();
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
