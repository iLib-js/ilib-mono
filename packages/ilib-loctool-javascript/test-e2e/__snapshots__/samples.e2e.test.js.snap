// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples js should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/t1.js" source-language="en-US" product-name="sample-js">
    <body>
      <trans-unit id="1" resname="Hello" restype="string" datatype="javascript">
        <source>Hello</source>
      </trans-unit>
      <trans-unit id="2" resname="This is a string that has no translation yet" restype="string" datatype="javascript">
        <source>This is a string that has no translation yet</source>
      </trans-unit>
      <trans-unit id="3" resname="Thank you" restype="string" datatype="javascript">
        <source>Thank you</source>
      </trans-unit>
      <trans-unit id="4" resname="id1" restype="string" datatype="javascript">
        <source>This is a test with a unique id</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`samples js-json should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/t1.js" source-language="en-US" product-name="sample-js-json">
    <body>
      <trans-unit id="1" resname="Hello" restype="string" datatype="javascript">
        <source>Hello</source>
      </trans-unit>
      <trans-unit id="2" resname="This is a string that has no translation yet" restype="string" datatype="javascript">
        <source>This is a string that has no translation yet</source>
      </trans-unit>
      <trans-unit id="3" resname="Thank you" restype="string" datatype="javascript">
        <source>Thank you</source>
      </trans-unit>
      <trans-unit id="4" resname="id1" restype="string" datatype="javascript">
        <source>This is a test with a unique id</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`samples php-resource should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/t1.js" source-language="en-US" product-name="sample-php-resource">
    <body>
      <trans-unit id="1" resname="Hello" restype="string" datatype="javascript">
        <source>Hello</source>
      </trans-unit>
      <trans-unit id="2" resname="This is a string that has no translation yet" restype="string" datatype="javascript">
        <source>This is a string that has no translation yet</source>
      </trans-unit>
      <trans-unit id="3" resname="Thank you" restype="string" datatype="javascript">
        <source>Thank you</source>
      </trans-unit>
      <trans-unit id="4" resname="id1" restype="string" datatype="javascript">
        <source>This is a test with a unique id</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
